tests/test_theming/test_theming.py

```diff
@@ -126,3 +149,41 @@ def test_theme_sidebars(app, status, warning):
     assert '<h3>Related Topics</h3>' not in result
     assert '<h3>This Page</h3>' not in result
     assert '<h3 id="searchlabel">Quick search</h3>' in result
+
+
+@pytest.mark.parametrize(
+    'theme_name',
+    [
+        'alabaster',
+        'agogo',
+        'basic',
+        'bizstyle',
+        'classic',
+        'default',
+        'epub',
+        'haiku',
+        'nature',
+        'nonav',
+        'pyramid',
+        'scrolls',
+        'sphinxdoc',
+        'traditional',
+    ],
+)
+def test_theme_builds(make_app, rootdir, sphinx_test_tempdir, theme_name):
+    """Test all the themes included with Sphinx build a simple project and produce valid XML."""
+    testroot_path = rootdir / 'test-basic'
+    srcdir = sphinx_test_tempdir / f'test-theme-{theme_name}'
+    shutil.copytree(testroot_path, srcdir)
+
+    app = make_app(srcdir=srcdir, confoverrides={'html_theme': theme_name})
+    app.build()
+    assert not app.warning.getvalue().strip()
+    assert app.outdir.joinpath('index.html').exists()
+
+    # check that the generated HTML files are well-formed (as strict XML)
+    for html_file in app.outdir.rglob('*.html'):
+        try:
+            xml_parse(html_file)  # noqa: S314
```

Comment:
FYI @picnixz @jayaddison, got this interesting lint error:

```
tests/test_theming/test_theming.py:187:13: S314 Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents
```

I guess in this case its ok to ignore?

Reply:
My opinion is that we should use `defusedxml` -- safer for future contributors when they create/open branches and pull requests.

Reply:
(I didn't in my PR either, admittedly! - but after becoming aware of this, I should've done)

Reply:
Fine by me, so will add `https://pypi.org/project/defusedxml/` to the testing dependencies

Reply:
added

Reply:
Honestly, I'm not for adding another dependency because the `etree_parse` of `sphinx.testing.util` uses the native API and we suppress the error code.

Reply:
ahaha just added, but ok I can remove again 😅 
@jayaddison is that ok with you?

Reply:
On behalf of the administrators of servers running Sphinx tests during package builds / assurance tests, and developers running them while developing random features and suggestions from colleagues: please use `defusedxml` :)

Reply:
There are two schools:

- either you trust contributors and their reviewers
- you don't trust them and we add the dependency (but I'm not sure if it is fully supported for all python versions and how well it is supported on 3.13 which updates quite often)

Reply:
I'm going fo lunch, I'll let you guys hash it out 😅

Reply:
Ok I forgot about people suggesting random addition or stackoverflow c/c. I am convinced! (let's keep it)